rust_src/src/alloc.rs

```diff
@@ -112,4 +114,20 @@ pub fn purecopy(obj: LispObject) -> LispObject {
     }
 }
 
+/// Return a newly allocated marker which does not point to any place.
+#[lisp_fn(c_name = "make_marker", name = "make-marker")]
+pub fn make_marker_lisp() -> LispObject {
+    let val: LispObject = unsafe { allocate_misc(Lisp_Misc_Type::Lisp_Misc_Marker) };
+    let mut p: LispMarkerRef = val.as_marker().unwrap();
```

Comment:
These two lines can be written as
```rust
let mut p = unsafe { allocate_misc(Lisp_Misc_Type::Lisp_Misc_Marker) }.force_marker();
```
If `force_marker` doesn't all ready exist, add these methods to the first `impl LispObject` you see in `marker.rs`:
```rust
    pub unsafe fn to_marker_unchecked(self) -> LispMarkerRef {
        LispMarkerRef::new(self.get_untaggedptr() as *mut Lisp_Marker)
    }

    pub fn force_marker(self) -> LispMarkerRef {
        unsafe { self.to_marker_unchecked() }
    }
```

Reply:
There is only one thing I am unsure of regarding this. Should I change the return type of the function to `LispMarkerRef`? Thanks in advance!

Reply:
Yep, generally we want return types to be as specific as possible